src/v/cluster/tests/partition_balancer_planner_test.cc

```diff
@@ -0,0 +1,535 @@
+ *   node_4: partitions: 2;
+ *   node_5: partitions: 2;
+ */
+FIXTURE_TEST(
+  test_move_multiple_partitions_batched_node_overfill,
+  partition_balancer_planner_fixture) {
+    vlog(logger.debug, "test_move_multiple_partitions_batched_node_overfill");
+    allocator_register_nodes(3);
+    create_topic("topic-1", 3, 3);
+    allocator_register_nodes(3);
+
+    auto hr = create_health_report();
+    auto fm = create_follower_metrics();
+
+    // Set full nodes
+    hr.node_reports[0].local_state.disks[0].free = full_node_free_size;
+    hr.node_reports[1].local_state.disks[0].free = full_node_free_size;
+    hr.node_reports[2].local_state.disks[0].free = full_node_free_size;
+
+    auto reassignments = planner.get_ntp_reassignments(hr, fm);
+    BOOST_REQUIRE_EQUAL(reassignments.size(), 2);
+
+    std::unordered_set<model::node_id> expected_nodes(
+      {model::node_id(3), model::node_id(4), model::node_id(5)});
+
+    auto new_replicas_1
+      = reassignments[0].allocation_units.get_assignments().front().replicas;
+    BOOST_REQUIRE_EQUAL(new_replicas_1.size(), expected_nodes.size());
+    check_expected_assignments(new_replicas_1, expected_nodes);
+
+    auto new_replicas_2
+      = reassignments[1].allocation_units.get_assignments().front().replicas;
+    check_expected_assignments(new_replicas_2, expected_nodes);
+}
+
+/*
+ * 5 nodes; 1 topic; 1 node down; 1 node full;
+ * Can move only 2 topics by one operation
+ * Actual
+ *   node_0: partitions: 1; down: True; disk: unfilled;
+ *   node_1: partitions: 1; down: False; disk: full;
+ *   node_2: partitions: 1; down: False; disk: unfilled;
+ *   node_3: partitions: 0; down: False; disk: unfilled;
+ *   node_4: partitions: 0; down: False; disk: unfilled;
+ * Expected
+ *   node_0: partitions: 0;
+ *   node_1: partitions: 0;
+ *   node_2: partitions: 1;
+ *   node_3: partitions: 1;
+ *   node_4: partitions: 1;
+ */
+FIXTURE_TEST(
+  test_one_node_down_one_node_full, partition_balancer_planner_fixture) {
+    vlog(logger.debug, "test_one_node_down_one_node_full");
+    allocator_register_nodes(3);
+    create_topic("topic-1", 1, 3);
+    allocator_register_nodes(2);
+
+    auto hr = create_health_report();
+
+    std::set<size_t> unavailable_nodes = {0};
+    auto fm = create_follower_metrics(unavailable_nodes);
+
+    // Set one node full
+    hr.node_reports[1].local_state.disks[0].free = full_node_free_size;
+
+    auto reassignments = planner.get_ntp_reassignments(hr, fm);
+    BOOST_REQUIRE_EQUAL(reassignments.size(), 1);
+
+    std::unordered_set<model::node_id> expected_nodes(
+      {model::node_id(2), model::node_id(3), model::node_id(4)});
+
+    auto new_replicas
+      = reassignments[0].allocation_units.get_assignments().front().replicas;
+    check_expected_assignments(new_replicas, expected_nodes);
+}
+
+/*
+ * 3 nodes; 2 topic; 2 node full;
+ * Movement should be from more filled node
+ * Actual
+ *   node_0: topics: topic-1; partitions: 3; down: True; disk: full;
```

Comment:
how can we move single-replica partitions from a node when it is down?

Reply:
It was a typo node is not down